### PR TITLE
Move from Azure SQL Edge to native SQL Server

### PR DESCRIPTION
**Pull request by mzieniukbw**
## 🎟️ Tracking

N/A

## 📔 Objective

Move from Azure SQL Edge to Microsoft SQL Server.
This is due to ARM platform problems on M1 macs for Azure SQL Edge. The database keeps crashing without any reason.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation
  team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: docs/getting-started/server/guide.md [hunk 1]
```diff
@@ -134,10 +134,10 @@ After that, rerun the docker compose command from Step 5.
...
 ### SQL Server
 
-In order to support ARM based development environments such as the M1 Macs, we use the
-[Azure SQL Edge](https://hub.docker.com/_/microsoft-azure-sql-edge) docker container instead of a
-normal [Microsoft SQL Server](https://hub.docker.com/_/microsoft-mssql-server) container. It behaves
-mostly identical to a regular SQL Server instance and runs on port 1433.
+For ARM based development environments such as the M1 Macs, we use the
+[Microsoft SQL Server](https://hub.docker.com/_/microsoft-mssql-server) docker container with
+`linux/amd64` platform emulation. It behaves identical to a SQL Server instance without emulation
+and runs on port 1433.
```
Comment (Hinton) [inline docs/getting-started/server/guide.md:140]: I think we can remove most of this text since we're back on the original docker container.

Comment (mzieniukbw) [reply docs/getting-started/server/guide.md]: Fixed in [baf1ccd](https://github.com/bitwarden/contributing-docs/pull/435/commits/baf1ccde2abec95f32974b67a77bca3a9a15a49d)

**Comment by withinfocus** [comment]
We don't do this though: https://github.com/bitwarden/server/blob/f72932bf2496e70776a5fb81f0d4734b3807da9d/dev/docker-compose.yml#L5

Can you share some logs, perhaps in our DBOps channel? I'd like to work with Microsoft instead of dropping Azure SQL Edge as it's a much smaller footprint.

**Comment by mzieniukbw** [comment]
@withinfocus Azure SQL Edge for ARM  is EOL and the bug seems to be on Microsoft image, only surfaced after linux kernel update on Docker Desktop, see more details [here](https://github.com/docker/for-mac/issues/7368#issuecomment-2275555064).
I think moving to Microsoft SQL Server is the only supported option onwards.
Also note, Azure SQL Edge on `linux/amd64` platform have not been updated for a while, unclear if and when it will get the fix.

**Review by withinfocus — APPROVED** [review]
Crazy, I didn't know about https://techcommunity.microsoft.com/t5/sql-server-blog/azure-sql-edge-update-september-2023/ba-p/3930827 -- I guess we need to move.

**Review by jonashendrickx — APPROVED** [review]

**Pull request merged**